exporter/awsemfexporter/metric_translator.go

```diff
@@ -118,6 +134,268 @@ func (dps DoubleHistogramDataPointSlice) At(i int) DataPoint {
 	return dps.DoubleHistogramDataPointSlice.At(i)
 }
 
+
+// Retrieve namespace for given set of metrics from user config 
+func getNamespace(rm *pdata.ResourceMetrics, namespace string) (string) {
+	if len(namespace) == 0 && !rm.Resource().IsNil() {
+		serviceName, svcNameOk := rm.Resource().Attributes().Get(conventions.AttributeServiceName)
+		serviceNamespace, svcNsOk := rm.Resource().Attributes().Get(conventions.AttributeServiceNamespace)
+		if svcNameOk && svcNsOk && serviceName.Type() == pdata.AttributeValueSTRING && serviceNamespace.Type() == pdata.AttributeValueSTRING {
+			namespace = fmt.Sprintf("%s/%s", serviceNamespace.StringVal(), serviceName.StringVal())
+		} else if svcNameOk && serviceName.Type() == pdata.AttributeValueSTRING {
+			namespace = serviceName.StringVal()
+		} else if svcNsOk && serviceNamespace.Type() == pdata.AttributeValueSTRING {
+			namespace = serviceNamespace.StringVal()
+		}
+	}
+
+	if len(namespace) == 0 {
+		namespace = defaultNameSpace
+	}
+	return namespace
+}
+
+// TranslateOtToGroupedMetric converts OT metrics to GroupedMetric format
+func TranslateOtToGroupedMetric(metric pdata.Metrics, config *Config) (map[string]*GroupedMetric, int) {
```

Comment:
It looks like this function is just a merging of `TranslateCWMetricToEMF`, `getCWMetrics`, and `buildCWMetric`. Instead of doing this, I would recommend leaving the logic as is and modify only the part you need (i.e. pass in `groupedMetricMap`). I suggest this for the following reasons:
1. Easier to understand which parts you changed (vs copied from other functions)
2. The test cases are already implemented for those functions. Putting them altogether would make them redundant, unless you merge all those test cases for this function
3. I prefer breaking down functions into smaller, more granular sub-functions as it is easier to read and test